MonoZelda/Commands/CollisionCommands/PlayerItemCollisionCommand.cs

```diff
@@ -0,0 +1,45 @@
+﻿using MonoZelda.Collision;
+using MonoZelda.Controllers;
+using MonoZelda.Sprites;
+
+namespace MonoZelda.Commands.CollisionCommands;
+
+public class PlayerItemCollisionCommand : ICommand
+{
+    private MonoZeldaGame game;
+
+    public PlayerItemCollisionCommand()
+    {
+        //empty
+    }
+
+    public PlayerItemCollisionCommand(MonoZeldaGame game)
+    {
+        this.game = game;
+    }
+
+    public void Execute(params object[] metadata)
+    {
+        Collidable collidableA = (Collidable) metadata[0];
+        Collidable collidableB = (Collidable) metadata[1];
+        CollisionController collisionController = (CollisionController) metadata[2];
+
+        if (collidableA.type == CollidableType.Item)
+        {
+            SpriteDict collidableDict = collidableA.CollidableDict;
+            collidableDict.Enabled = false;
+            collisionController.RemoveCollidable(collidableA);
+        }
+        else
+        {
+            SpriteDict collidableDict = collidableB.CollidableDict;
+            collidableDict.Enabled = false;
+            collisionController.RemoveCollidable(collidableB);
+        }
+    }
```

Comment:
So, does this code just destroy the item? It doesn't actually equip the player with the item or anything?

Reply:
Yeah, I am not sure, but I think after the item is destroyed, we would have to update Link's inventory, which would like a inventory class, and update that link has picked up the item in the GUI of the game.